lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java

```diff
@@ -3248,6 +3261,283 @@ void should_ReturnEmptyList_When_QueryingForNonexistentSystemInstanceLike() {
+                Triplet.of("CustomInt2", 2, TaskCustomIntField.CUSTOM_INT_2),
+                Triplet.of("CustomInt3", 3, TaskCustomIntField.CUSTOM_INT_3),
+                Triplet.of("CustomInt4", 4, TaskCustomIntField.CUSTOM_INT_4),
+                Triplet.of("CustomInt5", 5, TaskCustomIntField.CUSTOM_INT_5),
+                Triplet.of("CustomInt6", 6, TaskCustomIntField.CUSTOM_INT_6),
+                Triplet.of("CustomInt7", 7, TaskCustomIntField.CUSTOM_INT_7),
+                Triplet.of("CustomInt8", 8, TaskCustomIntField.CUSTOM_INT_8));
+
+        ThrowingConsumer<Triplet<String, Integer, TaskCustomIntField>> test =
+            t -> {
+              List<TaskSummary> result =
+                  taskService
+                      .createTaskQuery()
+                      .workbasketIdIn(wb.getId())
+                      .customIntAttributeIn(t.getRight(), t.getMiddle())
+                      .list();
+              assertThat(result).containsExactly(taskSummary1);
+            };
+
+        return DynamicTest.stream(testCases.iterator(), Triplet::getLeft, test);
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @TestFactory
+      Stream<DynamicTest> should_ApplyFilter_When_QueryingForCustomIntNotIn() {
+        List<Triplet<String, Integer, TaskCustomIntField>> testCases =
+            List.of(
+                Triplet.of("CustomInt1", 1, TaskCustomIntField.CUSTOM_INT_1),
+                Triplet.of("CustomInt2", 2, TaskCustomIntField.CUSTOM_INT_2),
+                Triplet.of("CustomInt3", 3, TaskCustomIntField.CUSTOM_INT_3),
+                Triplet.of("CustomInt4", 4, TaskCustomIntField.CUSTOM_INT_4),
+                Triplet.of("CustomInt5", 5, TaskCustomIntField.CUSTOM_INT_5),
+                Triplet.of("CustomInt6", 6, TaskCustomIntField.CUSTOM_INT_6),
+                Triplet.of("CustomInt7", 7, TaskCustomIntField.CUSTOM_INT_7),
+                Triplet.of("CustomInt8", 8, TaskCustomIntField.CUSTOM_INT_8));
+
+        ThrowingConsumer<Triplet<String, Integer, TaskCustomIntField>> test =
+            t -> {
+              List<TaskSummary> result =
+                  taskService
+                      .createTaskQuery()
+                      .workbasketIdIn(wb.getId())
+                      .customIntAttributeNotIn(t.getRight(), t.getMiddle())
+                      .list();
+              assertThat(result).containsExactlyInAnyOrder(taskSummary2);
+            };
+
+        return DynamicTest.stream(testCases.iterator(), Triplet::getLeft, test);
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @TestFactory
+      Stream<DynamicTest> should_ApplyFilter_When_QueryingForCustomIntWithin() {
+        List<Quadruple<String, TaskCustomIntField, IntInterval, IntInterval>> testCases =
+            List.of(
+                Quadruple.of(
+                    "CustomInt1",
+                    TaskCustomIntField.CUSTOM_INT_1,
+                    new IntInterval(1, 1),
+                    new IntInterval(-10, -8)),
```

Comment:
* replace string with enum for descrition
* user IntInterval[]

Reply:
Ok :)